### PR TITLE
[WIP] Evaluation

### PR DESCRIPTION
**Pull request by jorshi**
Starting on the evaluation runner. Metric for a task are defined in the task metadata and are automatically run for each embedding / task.

### CODE REVIEW

File: heareval/predictions/task_predictions.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def __len__(self):
         # return len(self.labels)
 
     def __getitem__(self, idx):
-        return self.embedding_memmap[idx]
+        return np.copy(self.embedding_memmap[idx])
```
Comment (jorshi) [inline heareval/predictions/task_predictions.py:78]: #142 -- added this to try. Can keep it if we want. Probably needs to be profiled.

File: heareval/evaluation/task_evaluation.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+def task_evaluation(task_path: Path):
+
+    metadata = json.load(task_path.joinpath("task_metadata.json").open())
+    label_vocab = pd.read_csv(task_path.joinpath("labelvocabulary.csv"))
+
+    embedding_type = metadata["embedding_type"]
+
+    if "evaluation" not in metadata:
+        print(f"Task {task_path.name} has no evaluation config.")
```
Comment (jorshi) [inline heareval/evaluation/task_evaluation.py:61]: This should raise an error

File: heareval/evaluation/runner.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+from heareval.evaluation.task_evaluation import task_evaluation
+
+
+@click.command()
+@click.option(
+    "--embeddings-dir",
+    default="embeddings",
+    help="Location of task embeddings to compute evaluation on.",
+    type=click.Path(exists=True),
+)
```
Comment (turian) [inline heareval/evaluation/runner.py:21]: Wait why don't we need the package we're evaluating here? hearbaseline etc.

Comment (jorshi) [reply heareval/evaluation/runner.py:21]: Why not compute evaluation on all them at once? They are all subfolders in the embeddings dir. Can add an option to specify and just run one.

File: heareval/evaluation/task_evaluation.py [hunk 4]
```diff
@@ -10,91 +10,123 @@
...
+        for i, prediction in enumerate(predictions):
+            predicted_class = np.argmax(prediction)
+            assert len(targets[i]) == 1
+            target_class = label_vocab[targets[i][0]]
+
+            if predicted_class == target_class:
+                correct += 1
+
+        error = correct / len(targets)
+        return {"top1_error": error}
```
Comment (khumairraj) [inline heareval/evaluation/task_evaluation.py:74]: `sklearn.metrics` has a `top_k_accuracy_score`. We can use that `k==1`

Comment (jorshi) [reply heareval/evaluation/task_evaluation.py:74]: I like the idea of using the sklearn metric

File: heareval/evaluation/runner.py (hunk 3 above)
Comment (turian) [reply heareval/evaluation/runner.py:21]: True.

**Comment by jorshi** [comment]
I think this can be closed once #165 is merged?

**Comment by turian** [comment]
#165 pulled this in
